src/utilities/getCurrentDate.ts

```diff
@@ -69,17 +73,35 @@ export function generateDateOrig(currentDate: DateData): Date {
 }
 
 export function generateWeekList(currentDate: DateData): WeekDay[] {
+	const currentDateCopy: DateData = {
```

Comment:
Deep clone to prevent passed in useState to be changed. This way we can control when to change the selectedDate more specifically. This is what was causing the next/prev day buttons to skip weeks.